wp-api.js

```diff
@@ -143,7 +128,7 @@
 			});
 
 			// Parse the author into a User object
-			if ( response.author !== 'undefined' ) {
+			if ( 'undefined' typeof response.author ) {
```

Comment:
Syntax error: should be `'undefined' === typeof response.author`, as it is below. Although since we have Underscore I'd probably `_.isUndefined()`